### PR TITLE
Update according to the recent proposal progress

### PR DESCRIPTION
**Pull request by afmenez**
Updating the es2016+ and esnext tables according to the recent proposal progress/changes.

Closes #1300

### CODE REVIEW

**Comment by afmenez** [comment]
This PR supercedes #1300

**Comment by zloirock** [comment]
`Set` methods were renamed.

File: data-esnext.js [hunk 1]
```diff
@@ -2691,20 +2691,28 @@ exports.tests = [
   ],
   subtests: [
     {
-      name: 'Array.prototype.flatten',
+      name: 'Array.prototype.flat',
       mdn: 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatten',
```
Comment (ljharb) [inline data-esnext.js]: Can we rename the mdn article too?

**Review by ljharb — APPROVED** [review]
Pending set methods renaming

**Comment by afmenez** [comment]
Done Set methods renaming.

**Review by zloirock — APPROVED** [review]

**Pull request merged**